### ISSUE TITLE
Consume Result always null

### ISSUE THREAD

**Issue by tarim** [confluentinc/confluent-kafka-dotnet#969]
Description
===========

This is a my first time using Kafka consumer to consume some topic on the Kafka cluster. I did same as example, but when I do consume(`var consumeResult = consumer.Consume(TimeSpan.FromSeconds(5));`) it always null. If I do with cancellation token
`var consumeResult = consumer.Consume(cts.Token);` It blocks and never return. I don't know, what I am doing wrong.  

Thank you,

My configuration 
` <kafka>
    <add key="group.id" value="grp_kafkadev" />
    <add key="sasl.username" value="kafkadev" />
    <add key="sasl.password" value="SKF1j5aA9fMALuDDWvI5pZAM7s=" />
    <add key="ssl.ca.location" value="C:\temp\DigiCertCA.crt" />
    <add key="enable.partition.eof" value="true" />
    <add key="sasl.mechanism" value="PLAIN" />
    <add key="security.protocol" value="sasl_ssl" />
      <add key="enable.auto.commit" value="false" />
    <add key="auto.commit.interval.ms" value="5000" />
    <add key="debug" value="all" /> 
    <add key="auto.offset.reset" value="earliest" /> 
      <add key="bootstrap.servers" value="data01.web.com:9093" />
    </kafka>`

How to reproduce
================

Checklist
=========

Please provide the following information:
 - [ ] A complete (i.e. we can run it), minimal program demonstrating the problem. No need to supply a project file.
 - [x] Confluent.Kafka nuget version.
 - [ ] Apache Kafka version.
 - [x] Client configuration.
 - [ ] Operating system.
 - [ ] Provide logs (with "debug" : "..." as necessary in configuration).
 - [ ] Provide broker log excerpts.
 - [ ] Critical issue.

**Comment by mhowlett** [comment]
there are either no messages in the topic, or you're at the end of the topic. you can confirm this by setting `EnablePartitionEOF` to true and seeing if you get any eof `ConsumeResult`s. note that by default when you first start consuming with a new consumer group, the start offset will be determined by `AutoOffsetReset`, which by default is `Latest`. You might need to set that to `Earliest`.

**Comment by roederja2** [comment]
I think returning null is not very nice. The example code doesn't even show that you need to check for this. It should either throw an exception or, preferably, return a result indicating that there is no message - reading your answer above this behavior can be turned on with the `EnablePartitionEOF` setting?

**Comment by mhowlett** [comment]
the examples all use the variant of `Consume` that takes a cancellation token, which never returns `null` (it will block until there is a message). this is the preferred pattern. It's not appropriate for `Consume(TimeSpan)` to throw an exception if there isn't a message since this isn't an exceptional scenario. If you enable partition eof, you'll get a `ConsumeResult` with IsPartitionEof true when the consumer reaches the end of partition. It would be arguably better to return a `ConsumeResult` indicating no message available if that is the case, though a downside of this is it will increase pressure on GC and may meaningfully reduce performance. It would be breaking to change this now, so we can't realistically consider it.

**Comment by cwaldron** [comment]
I'm running the example, using the Consume method and I'm running a debugger into Consumer.cs.  Consume eventually calls kafkaHandle.ConsumerPoll which returns an IntPtr.Zero meaning a null ConsumeResult is returned.  The outer Consume call remains in the while loop until it receives a non null ConsumeResult or a cancellation occurs.  I've set `EnablePartitionEOF` to true.  Therefore this is not the issue in the case I'm seeing.  

I've used Kafka.Net * KafkaSharp and have no issues the Consumer receiving message.  This only having this issue with Confluent.Kafka.
